src/common/logging.cpp

```diff
@@ -56,14 +56,17 @@ void createLoggers(const marian::Config* options) {
   std::vector<std::string> generalLogs;
   std::vector<std::string> validLogs;
 
-  if(options && options->has("log")) {
+  if(options && options->has("log")
+     && !options->get<std::string>("log").empty()) {
```

Comment:
Should the empty string be an allowed "disable" value for this? Why not just not specify it? If this is the only way to find out whether the variable is not specified (because it is declared as defaulting to the empty string), then consider to add a function next to has() that contains the above expression, and expresses this in its name.

Reply:
It should be pointing to a file location. I think an empty log should even abort? 
@snukky Is there a setting for that says if set it should be given with a value?

Reply:
The extended condition was necessary before I introduced `add_nondefault`. It's not needed anymore, so I reverted the change.